src/System.Text.Utf8/src/System/Text/Utf8/Utf8StringEnumerator.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Text.Utf8
+{
+    // TODO: This object is heavier than string itself... Once we got ByteSpan runtime support, change it.
+    public struct Utf8StringEnumerator : IEnumerator<UnicodeCodePoint>
```

Comment:
This type should be nested in Utf8String and should be called Enumerator. This is the pattern we use. See List<T>